examples/Consumer/Program.cs

```diff
@@ -115,23 +113,14 @@ public static void Run_Consume(string brokerList, List<string> topics, Cancellat
                             }
 
                             Console.WriteLine($"Received message at {consumeResult.TopicPartitionOffset}: {consumeResult.Message.Value}");
-
-                            if (consumeResult.Offset % commitPeriod == 0)
+                            try
+                            {
+                                // StoreOffset stores offset in an in-memory store which will be automatically committed every AutoCommitIntervalMs.
```

Comment:
maybe:

"Store the offset associated with `consumeResult` to a local cache. Stored offsets are committed to Kafka by a background thread every `AutoCommitIntervalMs`. The offset stored is actually the offset of the consumeResult + 1 since by convention, committed offsets specify the next message to consume. If `EnableAutoOffsetStore` had been set to the default value `true`, the .NET client would automatically store offsets immediately prior to delivering messages to the application. Explicitly storing offsets after processing gives at-least once semantics, the default behavior does not."

Reply:
we really should also document the Erroneous state gotcha here...

Reply:
Thanks, Updated the comment.

Reply:
@mhowlett For Erroneous state above comment already specifies this will ensure at least once consumption. Did you mean to document something other than that as well here?